gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -275,17 +280,30 @@ protected MRCompactorJobRunner(Dataset dataset, FileSystem fs) {
   public void run() {
     Configuration conf = HadoopUtils.getConfFromState(this.dataset.jobProps());
 
-    // Turn on mapreduce output compression by default
-    if (conf.get("mapreduce.output.fileoutputformat.compress") == null && conf.get("mapred.output.compress") == null) {
-      conf.setBoolean("mapreduce.output.fileoutputformat.compress", true);
-    }
+    try {
```

Comment:
Why not set the mapreduce.output.fileoutputformat.compress* config directly in the job properties which should be propagated to the Configuration object above?

Reply:
If property name is same and is defined at both the places, job config and hadoop config, I have observed that property's value from the latter over rides the former. For this reason I have defined new properties by prefixing "compaction.job" to mapreduce.output.fileoutputformat.compress*.

Reply:
How are you running the compaction job? Are you using MRCompactionRunner or CompactionRunner?

Reply:
I use**gobblin-compaction.sh**. with type as "mr".  I pass following class parameters as part of job config file
**compaction.jobprops.creator.class=gobblin.compaction.mapreduce.MRCompactorTimeBasedJobPropCreator
compaction.job.runner.class=gobblin.compaction.mapreduce.avro.MRCompactorAvroKeyDedupJobRunner
compaction.datasets.finder=gobblin.compaction.dataset.TimeBasedSubDirDatasetsFinder**

Reply:
That will use the MRCompactionRunner. The main of that class will call CliOptions.parseArgs(). This is where the job config is being overwritten with the hadoop configuration. I think this is wrong. The proper fix is to change parseArgs to only add the hadoop config into the job config if it is not already set.